demisto_sdk/commands/format/update_generic_json.py

```diff
@@ -114,6 +111,8 @@ def remove_spaces_end_of_id_and_name(self):
                 """
         if not self.old_file:
             if self.verbose:
-                click.echo('Updating YML ID and name to be without spaces at the end')
-            self.data['name'] = self.data['name'].strip()
-            self.data['id'] = self.data['id'].strip()
+                click.echo('Updating json ID and name to be without spaces at the end')
+            if 'name' in self.data:
+                self.data['name'] = self.data['name'].strip()
+            if 'id' in self.data:
+                self.data['id'] = self.data['id'].strip()
```

Comment:
First of all the echo command prints "Updating YML" while it is in update json.
I added some checks because some tests without name/Id encountered an error while trying to get this field. (This happens because we generalized some updates)

Reply:
can you specify for which types it happened?

Reply:
connection, old layout, old classifier